contracts/FiatToken.cdc

```diff
@@ -1,1287 +1,258 @@
-        blocklisterAccount.link<&Blocklister{ResourceId}>(self.BlocklisterUUIDPubPath, target: self.BlocklisterStoragePath)
-        blocklisterAccount.link<&Blocklister{BlocklisterCapReceiver}>(self.BlocklisterCapReceiverPubPath, target: self.BlocklisterStoragePath)
-
-        // Setup the Pauser
-        pubKeyAttrs = []
-        i = 0
-        while i < pauserPubKeys.length {
-            let pka = OnChainMultiSig.PubKeyAttr(sa: pauserPubKeysAlgos[i], w: pauserPubKeysWeights[i])
-            pubKeyAttrs.append(pka)
-            let key = PublicKey(
-                publicKey: pauserPubKeys[i].decodeHex(), 
-                signatureAlgorithm: SignatureAlgorithm(rawValue: pauserPubKeysAlgos[i]) ?? panic ("Invalid signature algo")
-            )
-            pauserAccount.keys.add(
-                publicKey: key,
-                hashAlgorithm: HashAlgorithm.SHA3_256,
-                weight: pauserPubKeysWeights[i]
-            )
-            i = i + 1
-        }
-        pauserAccount.save(<- self.createNewPauser(publicKeys: pauserPubKeys, pubKeyAttrs: pubKeyAttrs), to: self.PauserStoragePath)
-        pauserAccount.link<&Pauser{OnChainMultiSig.PublicSigner}>(self.PauserPubSigner, target: self.PauserStoragePath)
-        pauserAccount.link<&Pauser{ResourceId}>(self.PauserUUIDPubPath, target: self.PauserStoragePath)
-        pauserAccount.link<&Pauser{PauseCapReceiver}>(self.PauserCapReceiverPubPath, target: self.PauserStoragePath)
-
-        // Assign the admin capabilities
-        let adminExecutorRef = contractAccount.borrow<&FiatToken.AdminExecutor>(from: self.AdminExecutorStoragePath)
-            ?? panic("cannot borrow AdminExecutor from storage")
-        let ownerExecutorRef = contractAccount.borrow<&FiatToken.OwnerExecutor>(from: self.OwnerExecutorStoragePath)
-            ?? panic("cannot borrow OwnerExecutor from storage")
-        adminExecutorRef.changeAdmin(to: adminAccount.address, newPath: initialAdminCapabilityPrivPath)
-        ownerExecutorRef.reassignOwner(to: ownerAccount.address, newPath: initialOwnerCapabilityPrivPath)
-        ownerExecutorRef.reassignMasterMinter(to: masterMinterAccount.address, newPath: initialMasterMinterCapabilityPrivPath)
-        ownerExecutorRef.reassignBlocklister(to: blocklisterAccount.address, newPath: initialBlocklisterCapabilityPrivPath)
-        ownerExecutorRef.reassignPauser(to: pauserAccount.address, newPath: initialPauserCapabilityPrivPath) 
-
+ 
         // Create a Vault with the initial totalSupply
         let vault <- create Vault(balance: self.totalSupply)
-        self.account.save(<-vault, to: self.VaultStoragePath)
+        self.account.storage.save(<-vault, to: self.VaultStoragePath)
+ 
+        // Create a Vault with the initial totalSupply
+        let minter <- create MinterResource()
+        self.account.storage.save(<-minter, to: self.MinterStoragePath)
```

Comment:
Do you intend to make the contract account the minter?

Reply:
This code is purely for testing purposes. Since the contract has already been deployed on testnet and mainnet, this logic will never actually be executed. The minter resource will be stored in the EVM bridge so that only the bridge protocol will be able to mint tokens.

In my latest commit, I added an approximation of how the minter will be transferred to the bridge. We will likely have an `access(account)` [function](https://github.com/joshuahannan/flow-usdc/blob/feature/cadence-1/contracts/FiatToken.cdc#L216) in `FiatToken` that creates a minter, then we will deploy [a new contract](https://github.com/joshuahannan/flow-usdc/blob/feature/cadence-1/contracts/FiatTokenMinterToBridge.cdc) to the account that trustlessly can call this function and send the minter to the bridge without the minter ever being controlled by the FiatToken account or any other account.

The bridge design for this hasn't been fully finalized yet, but that should be roughly how getting the minter to it will work.

Does that make sense?

Reply:
Here is the current proposal for how `FiatToken` will fit into the bridge: https://github.com/onflow/flow-evm-bridge/pull/46#event-12582615999

Reply:
so you expect more changes to the contract once the bridge design is finalized?

Reply:
The `FiatToken` contract won't change at all (unless y'all request for specific changes), but the `FiatTokenMinterToBridge` contract might change depending on the exact method that we decide to use to deliver the minter, but I'm almost certain that the method I've shown here will be the method that is used.